text/0000-signed-transaction-payload/0000-signed-transaction-payload.md

```diff
@@ -0,0 +1,825 @@
+                                        <td>Signature</td>
+                                        <td>ByteArray</td>
+                                        <td>The signature signing the serialized <i>Unsigned Transaction</i>.</td>
+                                    </tr>
+                                </table>
+                            </details>
+                             <details>
+                                <summary>Ed25519 Signature</summary>
+                                <table>
+                                    <tr>
+                                        <th>Name</th>
+                                        <th>Type</th>
+                                        <th>Description</th>
+                                    </tr>
+                                    <tr>
+                                        <td>Signature Type</td>
+                                        <td>varint</td>
+                                        <td>
+                                            Set to <strong>value 1</strong> to denote an <i>Ed25519 Signature</i>.
+                                        </td>
+                                    </tr>
+                                    <tr>
+                                        <td>Public key</td>
+                                        <td>ByteArray[32]</td>
+                                        <td>The public key of the Ed25519 keypair which is used to verify the signature.</td>
+                                    </tr>
+                                    <tr>
+                                        <td>Signature</td>
+                                        <td>ByteArray[64]</td>
+                                        <td>The signature signing the serialized <i>Unsigned Transaction</i>.</td>
+                                    </tr>
+                                </table>
+                            </details>
+                        </td>
+                    </tr>
+                </table>
+            </details>
+            <details open="true">
+                <summary>Reference Unlock Block</summary>
+                <blockquote>
+                References a previous unlock block in order to substitute the duplication of the same unlock block data for inputs which unlock through the same data.
+                </blockquote>
+                <table>
+                    <tr>
+                        <th>Name</th>
+                        <th>Type</th>
+                        <th>Description</th>
+                    </tr>
+                    <tr>
+                        <td>Unlock Type</td>
+                        <td>varint</td>
+                        <td>
+                            Set to <strong>value 1</strong> to denote a <i>Reference Unlock Block</i>.
+                        </td>
+                    </tr>
+                    <tr>
+                        <td>Reference</td>
+                        <td>varint</td>
```

Comment:
why use varint if you set the rule to be < Count (max 127)? why not uint8?

Reply:
Using a fixed integer type here could give us a fixed byte size for the reference block, if that is desirable.